components/script/script_runtime.rs

```diff
@@ -447,6 +447,260 @@ pub struct Runtime {
     networking_task_src: Option<Box<NetworkingTaskSource>>,
 }
 
+impl Runtime {
+    /// Create a new runtime, optionally with the given [`NetworkingTaskSource`].
+    ///
+    /// # Safety
+    ///
+    /// If panicking does not abort the program, any threads with child runtimes will continue
+    /// executing after the thread with the parent runtime panics, but they will be in an
+    /// invalid and undefined state.
+    ///
+    /// This, like many calls to SpiderMoney API, is unsafe.
+    #[allow(unsafe_code)]
+    pub(crate) fn new(networking_task_source: Option<NetworkingTaskSource>) -> Runtime {
+        unsafe { Self::new_with_parent(None, networking_task_source) }
+    }
```

Comment:
safety section on safe fn, should this fn be marked as unsafe?

Reply:
Sometimes we put a safety section on functions that are marked safe when there are extra conditions, but do not require unsafe code blocks. I think it's just sort of a "nice to know" thing.